### PR TITLE
Upload image preview placeholder - Follow-up #7569

### PR DESCRIPTION
**Pull request by okansahin**
### Details

It's a follow-up PR of [7569](https://github.com/Expensify/App/pull/7569) to fixing show placeholder when image upload by drag&drop.

### Fixed Issues

$ https://github.com/Expensify/App/issues/7463

### Tests

- [x] Verify that no errors appear in the JS console

### Tests | QA Steps

1. Open a chat window
2. Click (+ button / add attachment / choose from gallery)
3. Select an image
4. Click Send button in preview modal

**AND**

1. Open a chat window
2. Drag an image from your computer to chat window
3. Click Send button in preview modal

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

##### Safari

https://user-images.githubusercontent.com/1966034/155301225-20e4774a-4152-4799-ae47-79af6c29f137.mov

##### Chrome

https://user-images.githubusercontent.com/1966034/155301249-2bc1e662-66dc-490a-a707-70bfcc1673ce.mov

#### Mobile Web

https://user-images.githubusercontent.com/1966034/155301270-2e2c5b70-7fa4-4966-81a4-bff7d532631d.mov

#### Desktop

https://user-images.githubusercontent.com/1966034/155301290-c451b42d-5786-4e66-b309-910fd216f9d0.mov

#### iOS

https://user-images.githubusercontent.com/1966034/155301311-27600604-7d75-40a3-a142-ded600b9c04c.mov

#### Android

https://user-images.githubusercontent.com/1966034/155301580-388f5d64-9535-45a3-a863-c146a9dfa04f.mov

### CODE REVIEW

**Comment by parasharrajat** [comment]
@okansahin Just add the original issue which this PR is concerned with. 
image

File: src/libs/actions/Report.js [hunk 1]
```diff
@@ -1138,7 +1138,16 @@ function addAction(reportID, text, file) {
     const parser = new ExpensiMark();
     const commentText = parser.replace(text);
     const isAttachment = _.isEmpty(text) && file !== undefined;
-    const attachmentInfo = isAttachment ? file : {};
+    let attachmentInfo = {};
```
Comment (parasharrajat) [inline src/libs/actions/Report.js:1141]: Just add source to the file object 
https://github.com/Expensify/App/blob/853db572e36748541ec6c809d12a3ff2b61be8b5/src/components/AttachmentModal.js#L115

``` _.extend(file, {source: this.state.sourceURL});```

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/libs/actions/Report.js [hunk 2]
```diff
@@ -1168,7 +1167,7 @@ function addAction(reportID, text, file) {
             ],
             isFirstItem: false,
             isAttachment,
-            attachmentInfo,
+            attachmentInfo: file,
```
Comment (parasharrajat) [inline src/libs/actions/Report.js:1170]:
```suggestion
            attachmentInfo: file,
```

Please revert this.

File: src/libs/actions/Report.js [hunk 3]
```diff
@@ -1138,7 +1138,16 @@ function addAction(reportID, text, file) {
     const parser = new ExpensiMark();
     const commentText = parser.replace(text);
     const isAttachment = _.isEmpty(text) && file !== undefined;
-    const attachmentInfo = isAttachment ? file : {};
```
[the remark this answers is not in the document]
Comment (parasharrajat) [reply src/libs/actions/Report.js:1105]: Revert this change.

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Please add proper QA steps. Another set for drag and drop feature.

**Comment by okansahin** [comment]
> Please add proper QA steps. Another set for drag and drop feature.

https://user-images.githubusercontent.com/1966034/155588680-67f3a00c-35c7-4786-bf3e-dcb2e7112f18.mov

**Comment by parasharrajat** [comment]
Thanks for the video @okansahin. I meant to add the QA steps for the drag and drop feature as well on the PR desktop under the QA section. 
image

These are meant for the QA team to test the PR. Thanks.

**Comment by okansahin** [comment]
### Tests | QA Steps
1. Open a chat window
2. Click (+ button / add attachment / choose from gallery)
3. Select an image
4. Click Send button in preview modal

and

1. Open a chat window
2. Drag an image from your computer to chat window
3. Click Send button in preview modal

**Comment by parasharrajat** [comment]
> on the PR description

**Comment by okansahin** [comment]
> > on the PR description

sure. done.

**Review by parasharrajat — APPROVED** [review]
LGTM. Tests well. 

@Julesssss 
:ribbon: :eyes: :ribbon:  C+ reviewed

**Review by Julesssss — APPROVED** [review]

**Comment by Julesssss** [comment]
Tests well for me 👍

**Comment by okansahin** [comment]
@Julesssss @JmillsExpensify 
Any news about regression period start?

**Comment by Julesssss** [comment]
Hi @okansahin, I'll comment in the linked issue.

**Comment by Stutikuls** [comment]
**Issue1 -**

**Title**- [Medium]: Chrome + Jaws: Screen Reader:  Role is not defined for the 'Send' control.
**Actual** - Role is not defined for the 'Send' control. Screen reader is reading "only send" when focus lands on the control.
**Expected** - Role - 'Button' should define for the 'Send' control and screen reader should read like "Send button".
**WCAG failure** - 4.1.2
**Reproducible in staging?:** - Yes
**Version Number**: - v1.1.42-2
**Platforms** - Desktop (MAC + Voiceover), Web(Chrome + Jaws), Mobile-web(iOS), iOS

**Note** - Due to installation of build not able to check on Android

https://user-images.githubusercontent.com/97600950/158092571-cab7623e-0daa-4a13-9f84-6cc33dc05bb9.mp4


**Issue 2 -**

**Title**- [Medium]: MAC : Screen Reader:  Name & Role is not defined for the 'Action (+)' control.
**Actual** - Name & Role is not defined for the 'Action (+)'  control. Screen reader is reading some extra information when focus lands on the Action (+) control.
**Expected** - Name = 'Action/Add (+) & Role - 'Button' should define for the 'Action (+)' control and screen reader should read like "Acton Add button".
**WCAG failure** - 4.1.2
**Reproducible in staging?:** - Yes
**Version Number**: - v1.1.42-2
**Platforms** - Desktop (MAC + Voiceover), Web(Chrome + Jaws)

**Note** - Due to installation of build not able to check on Android

https://user-images.githubusercontent.com/97600950/158092897-e7ff4bbd-1a19-4560-a090-e5dc1e190785.mp4

**Issue 3 -**

**Title**- [High]: Mobile - web (Safari + Voiceover) : 'Action (+)' control is not accessible using screen reader.
**Actual** - Action (+)' control is not accessible using screen reader.
**Expected** - Action (+)' control should accessible using screen reader.
**WCAG failure** - 2.1.1
**Reproducible in staging?:** - Yes
**Version Number**: - v1.1.42-2
**Platforms** - Mobile -web(Safari +Voiceover)

**Note** - Due to installation of build not able to check on Android

https://user-images.githubusercontent.com/97600950/158094306-2a0604dd-5b65-4b16-bba7-e7a1fd3ae333.mp4


**Issue 4 -**

**Title**- Functional : [Medium]: Chrome : Not able to drag the image using keyboard.
**Actual** - Not able to drag the image using keyboard. Using Mouse will do it but with keyboard not possible.
**Expected** - Some key combinations (like - Shift+ Down arrow key, etc.) should provide to drag the image using keyboard.
**WCAG failure** - Nil 
**Reproducible in staging?:** - Yes
**Version Number**: - v1.1.42-2
**Platforms** - Web (Chrome), Desktop (MAC)

https://user-images.githubusercontent.com/97600950/158094890-5088c386-5728-4840-b5f6-78c693993d1f.mp4

**Pull request merged**